### PR TITLE
Fixes: #517

### LINKED ISSUE #517: page_params is not defined
We are getting this error because on some pages the `page_params`is not there so we gotta add a check for the same.

### PR DESCRIPTION
**Pull request by aashish-ak**
---

**What's this PR do?**
  It fixes #517 ,

**Any background context you want to provide?**
 The error 'page_params is not defined' occured because when the third party authorization windows 
 are opened at the time of login, so i added a check that if the variable `page_params` exist only then load the DOM.

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

**Comment by abhigyank** [comment]
Hello @aashish-ak thanks for doing this!
However I think the correct approach could be to use a `try..catch` for the `serverLanguage` variable. I don't think hard-coding the websites is a good solution.

**Comment by aashish-ak** [comment]
Okay will it be better if i fix the hostname to `*.zulip.*` or `*.zulipchat.*` ?? Will it work?? @abhigyank

**Comment by abhigyank** [comment]
No we can't do that, as many people host Zulip on-premise and they have different domains.
Can you try something like?
```js
let serverLanguage = null;
try{
    serverLanguage = page_params.default_language;
} 
```

**Comment by aashish-ak** [comment]
@abhigyank this isn't helping as the the whole code in that block should be loaded if the window is not of a third party authorisation method.

**Comment by aashish-ak** [comment]
@abhigyank I've modified the code, merge it if you find the PR worthy :smile:

File: app/renderer/js/notification/index.js [hunk 1]
```diff
@@ -20,8 +20,14 @@ if (process.platform === 'darwin') {
...
 	// Call this function only when user is logged in
-	// eslint-disable-next-line no-undef, camelcase
-	if (page_params.realm_uri) {
+	let webpageParams = null;
+	try {
+		// eslint-disable-next-line no-undef, camelcase
+		webpageParams = page_params;
+	} catch (err) {
+		webpageParams = null;
+	}// eslint-disable-next-line no-undef, camelcase
```
Comment (abhigyank) [inline app/renderer/js/notification/index.js]: Can you move the comment to the new line?

Comment (aashish-ak) [reply app/renderer/js/notification/index.js]: @abhigyank  Done :+1:

**Comment by abhigyank** [comment]
@aashish-ak Looks good to me, left a comment about a small change.
Also can you squash both the commits into a single commit. You can look up `git rebase -i` for the same [here](https://zulip.readthedocs.io/en/latest/git/fixing-commits.html?highlight=squash#squashing-commits).
Also the can you change the commit message as per our guidelines [here](https://zulip.readthedocs.io/en/latest/contributing/version-control.html?highlight=commit%20message#commit-messages) . You can also look at other commit messages for references.

**Review by abhigyank — APPROVED** [review]

**Review by abhigyank — APPROVED** [review]
@akashnimare LGTM. Can be merged.

**Comment by akashnimare** [comment]
@aashish-ak, @abhigyank  since we are using the same code, I think we can wrap it up in a function and export it wherever we need, something like this -

```js
const isPageParams = () => {
    let webpageParams = null;
    try {
        webpageParams = page_params;
    } catch (err) {
        webpageParams = null;
    }
    return webpageParams;
}


module.exports = {
    isPageParams
}

```

**Comment by abhigyank** [comment]
Right, this should probably go in `default-util` right?

**Comment by akashnimare** [comment]
There or maybe we could create `params-util.js`.

**Comment by akashnimare** [comment]
@aashish-ak are you interested in finishing this PR?

**Comment by aashish-ak** [comment]
Sorry been a little busy, yes i'll finish it soon :smile:

**Comment by aashish-ak** [comment]
I'll complete the work till noon, you can merge easily after that ☺️

**Comment by akashnimare** [comment]
Awesome, would be great if you can finish it up since we're planning to add this in a new release.

**Comment by aashish-ak** [comment]
@akashnimare @abhigyank It's done, sorry for the late commit.

**Comment by akashnimare** [comment]
Merged after fixing linting errors and commit message. Thanks, @aashish-ak.

**Pull request closed without merging**